libs/wingc/src/nodeof_transform.rs

```diff
@@ -0,0 +1,77 @@
+use crate::{
+	ast::{CalleeKind, Expr, ExprKind, Reference, Symbol, UserDefinedType},
+	diagnostic::WingSpan,
+	fold::{self, Fold},
+};
+
+/// Transforms `nodeof(x)` expressions into `std.Node.of(x)`
+pub struct NodeofTransformer;
+
+impl NodeofTransformer {
+	pub fn new() -> Self {
+		Self
+	}
+}
+
+impl Fold for NodeofTransformer {
+	fn fold_expr(&mut self, expr: Expr) -> Expr {
+		let file_id = &expr.span.file_id;
+
+		let ExprKind::Call { callee, arg_list } = expr.kind else {
+			return fold::fold_expr(self, expr);
+		};
+
+		let arg_list = self.fold_args(arg_list);
+
+		let callee = match callee {
+			CalleeKind::Expr(call_expr) => call_expr,
+			_ => {
+				return Expr {
+					kind: ExprKind::Call { callee, arg_list },
+					id: expr.id,
+					span: expr.span,
+				};
+			}
+		};
+
+		let ExprKind::Reference(Reference::Identifier(Symbol { ref name, .. })) = callee.kind else {
```

Comment:
Does changing this line help with the inability to use `fold::fold_expr(self, expr)`? Probably not but I figured I'd shoot my shot.

```suggestion
		let ExprKind::Reference(Reference::Identifier(Symbol { name, .. })) = &callee.kind else {
```

Reply:
(removed this whole file - resolving)